server/services/User.js

```diff
@@ -15,3 +15,25 @@ export const createUser = (email, password) => {
     passwordSalt: salt,
   });
 };
+
+const isPasswordSame = (pass, encryptedPass, salt) => {
+  const passwordHash = getHmac(pass, salt + DB_SALT);
+
+  return passwordHash === encryptedPass;
+};
+
+export const getUser = (email, password) => {
+  return UserModel.findOne({ email }, (err, user) => {
+    if (!user) {
+      throw new Error('User not found');
+    }
+
+    const isPasswordValid = isPasswordSame(
```

Comment:
At the moment `getUser` method is responsible for 2 things at a time:

1. Retrieving the user
2. Checking if password is correct

So we should either rename method getUser to something else (e.g. _authorizeUser_) or simply move password checking logic up to the controller (this one is preferred)